File: mingachevir-real-estate-server-master/src/main/java/io/mingachevir/mingachevirrealestateserver/model/dto/HouseCardDto.java
package io.mingachevir.mingachevirrealestateserver.model.dto;

import io.mingachevir.mingachevirrealestateserver.model.entity.House;
import io.mingachevir.mingachevirrealestateserver.util.PriceType;
import lombok.Data;

import java.util.Date;

@Data
public class HouseCardDto {
    private Long id;
    private String name;
    private Double price;
    private PriceType priceType;
    private String location;
    private String coverImage;
    private Date createDate;

    public HouseCardDto(House house) {
        this.id = house.getId();
        this.name = house.getName();
        this.price = house.getPrice();
        this.priceType = house.getPriceType();
        this.location = house.getLocation();
        this.coverImage = house.getCoverImage();
        this.createDate = house.getCreateDate();
    }
}

File: mingachevir-real-estate-site-master/src/app/models/HouseCardDto.ts
import {PriceType} from "../util/PriceType";

export interface HouseCardDto {
  id: number;
  name: string;
  price: number | null;
  priceType: PriceType;
  location: string | null;
  coverImage: string | null;
  createDate: Date;
}
